site-src/geps/gep-718.md

```diff
@@ -0,0 +1,278 @@
+@@ -96,10 +96,7 @@ type GatewayReference struct {
+ 
+ // RouteForwardTo defines how a Route should forward a request.
+ type RouteForwardTo struct {
+-	// ServiceName refers to the name of the Service to forward matched requests
+-	// to. When specified, this takes the place of BackendRef. If both
+-	// BackendRef and ServiceName are specified, ServiceName will be given
+-	// precedence.
++	// BackendRef is a reference to a backend to forward matched requests to.
+ 	//
+ 	// If the referent cannot be found, the rule is not included in the route.
+ 	// The controller should raise the "ResolvedRefs" condition on the Gateway
+@@ -115,30 +112,16 @@ type RouteForwardTo struct {
+ 	// in the BackendPolicy resource. For custom backends, it is encouraged to
+ 	// add a semantically-equivalent field in the Custom Resource Definition.
+ 	//
+-	// Support: Core
+-	//
+-	// +optional
+-	// +kubebuilder:validation:MaxLength=253
+-	ServiceName *string `json:"serviceName,omitempty"`
+-
+-	// BackendRef is a reference to a backend to forward matched requests to. If
+-	// both BackendRef and ServiceName are specified, ServiceName will be given
+-	// precedence.
+-	//
+-	// If the referent cannot be found, the rule is not included in the route.
+-	// The controller should raise the "ResolvedRefs" condition on the Gateway
+-	// with the "DegradedRoutes" reason. The gateway status for this route should
+-	// be updated with a condition that describes the error more specifically.
+-	//
+-	// Support: Custom
++	// Support: Core (Kubernetes Service)
++	// Support: Custom (any other resource)
+ 	//
+ 	// +optional
++	// +kubebuilder:default={kind: "service"}
+ 	BackendRef *LocalObjectReference `json:"backendRef,omitempty"`
+ 
+ 	// Port specifies the destination port number to use for the
+-	// backend referenced by the ServiceName or BackendRef field.
+-	// If unspecified, the destination port in the request is used
+-	// when forwarding to a backendRef or serviceName.
++	// backend referenced by BackendRef field. Port is required when the backend
++	// is a Kubernetes Service.
+ 	//
+ 	// Support: Core
+ 	//
+```
+For Kubernetes Services, the API will look like the following:
+
+```yaml
+...
+forwardTo:
+- backendRef:
+    name: foo-service-v1
+  port: 80
+  weight: 80
+- backendRef:
+    name: foo-service-canary
+  port: 80
+  weight: 20
+...
+```
+
+Here, the `kind` field is omitted as it will be injected as a default.
+
+For custom backends, the API will look like the following:
+
+```yaml
+...
+forwardTo:
+- backendRef:
+    name: foo-v1
+    kind: server
+    group: networking.acme.io
+  port: 80
+  weight: 80
+- backendRef:
+    name: foo-v1-canary
+    kind: server
+    group: networking.acme.io
+  port: 80
+  weight: 20
+...
+```
+
+## Note on future consolidation
+
+It has been noted that `forwardTo` doesn't flow as good as `backendRefs`.
+We could consider further consolidating the YAML if API conventions allow for the
+following:
+
+```yaml
+backendRefs:
```

Comment:
The more I've thought about this, I actually think this approach could make the most sense. It's just as concise as the existing API while also being less confusing. I'm a little concerned that there will not be a way to reference/apply policy to a specific item in this list, but that seems like an edge case to me. In most cases that could be resolved by applying policy either to the route rule or to the backend itself.